boot/zephyr/uuid/uuid.c

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2018 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <bootutil/mcuboot_uuid.h>
+
+#define IMAGE_ID_COUNT CONFIG_UPDATEABLE_IMAGE_NUMBER
+#define CID_INIT(index, label) \
+	static const struct image_uuid label = {{ \
+		NCS_MCUBOOT_UUID_CID_IMAGE_## index ##_VALUE \
+	}}
+#define CID_CONFIG(index) UTIL_CAT(CONFIG_NCS_MCUBOOT_UUID_CID_IMAGE_, index)
+#define CID_DEFINE(index, prefix) \
+	IF_ENABLED(CID_CONFIG(index), (CID_INIT(index, prefix##index)))
+
+#define CID_CONDITION(index, label) \
+	if (image_id == index) { \
+		*uuid_cid = &label; \
+		FIH_RET(FIH_SUCCESS); \
+	}
+#define CID_CHECK(index, prefix) \
+	IF_ENABLED(CID_CONFIG(index), (CID_CONDITION(index, prefix##index)))
+
+static fih_ret boot_uuid_compare(const struct image_uuid *uuid1, const struct image_uuid *uuid2)
```

Comment:
not really inkeeping in the style of MCUboot - no other OS can use this, should be a base file in a common folder then the parts that zephyr needs in the zephyr specific file that gets included